modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -328,106 +337,289 @@ def find_esc3_vuln_cert_templates
+      print_error("Registry property '#{property_name}' not found at path '#{full_path}'.")
     end
-    user_groups = map_sids_to_names(user_groups)
+    value
+  end
 
-    # Determine what Certificate Templates are available to us
-    esc_raw_filter = '(objectclass=pkicertificatetemplate)'
+  def enum_registry_values
+    endpoint = "http://#{datastore['RHOST']}:5985/wsman"
+    user = datastore['LDAPUsername']
+    pass = datastore['LDAPPassword']
 
-    attributes = ['cn', 'description', 'ntSecurityDescriptor']
-    esc_entries = query_ldap_server(esc_raw_filter, attributes, base_prefix: CERTIFICATE_TEMPLATES_BASE)
+    conn = WinRM::Connection.new(
+      endpoint: endpoint,
+      user: user,
+      password: pass,
+      transport: :negotiate
+    )
 
-    return if esc_entries.empty?
+    # Check if registry values are already cached
+    cached_values = @ldap_objects.find { |obj| obj[:type] == :registry_values }
+    return cached_values[:values] if cached_values
 
-    # Determine if the user we've authenticated with has the ability to edit
-    esc_entries.each do |entry|
-      certificate_symbol = entry[:cn][0].to_sym
-      next if @certificate_details[certificate_symbol][:enroll_sids].empty?
+    registry_values = {}
+
+    begin
+      conn.shell(:powershell) do |shell|
+        registry_values[:certificate_mapping_methods] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Control\\SecurityProviders\\Schannel', 'CertificateMappingMethods')
+        registry_values[:strong_certificate_binding_enforcement] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\Kdc', 'StrongCertificateBindingEnforcement')
+
+        active_policy_name = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'Active')
+        registry_values[:disabled_extension_list] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'DisableExtensionList', active_policy_name)
+        registry_values[:edit_flags] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'EditFlags', active_policy_name)
+      end
+
+      if registry_values[:strong_certificate_binding_enforcement] == '1'
+        vprint_good('ESC9 could be exploitable using Kerberos due to StrongCertificateBindingEnforcement being set to 1')
+      elsif registry_values[:strong_certificate_binding_enforcement] == '0'
+        vprint_good('ESC10 could be exploitable using Kerberos due to StrongCertificateBindingEnforcement being set to 0')
+      elsif registry_values[:strong_certificate_binding_enforcement] == '2'
+        vprint_error('ESC9 and ESC10 cannot be exploited using Kerberos due to StrongCertificateBindingEnforcement being set to 2')
+      end
+
+      if registry_values[:certificate_mapping_methods] == '4' || registry_values[:certificate_mapping_methods] == '31'
```

Comment:
These are actually bitwise flags so there are more values than just 4 and 31 that would allow UPN mapping.
```suggestion
      if (registry_values[:certificate_mapping_methods].to_i & 4) > 0
```